src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -1229,5 +1229,69 @@ private static void CheckEffectiveAndDeducedBaseTypes(ConversionsBase conversion
                 conversions.HasIdentityOrImplicitReferenceConversion(deducedBase, effectiveBase, ref useSiteDiagnostics) ||
                 conversions.HasBoxingConversion(deducedBase, effectiveBase, ref useSiteDiagnostics));
         }
+
+        internal static TypeWithAnnotations MergeTopLevelNullabilityForConstraint(TypeWithAnnotations type1, TypeWithAnnotations type2)
```

Comment:
>Merge [](start = 44, length = 5)

nit: "Merge" did throw me off a bit because it has many connotations (Merge/Join in flow analysis). Maybe `StrictestTopLevelNullabilityConstraint`?